packages/dds/merge-tree/src/partialLengths.ts

```diff
@@ -483,34 +699,65 @@ export class PartialSequenceLengths {
 			return;
 		}
 
-		const firstGte = partials.firstGte(seqOrLocalSeq);
+		// overlapping move and remove, remove happened first
+		if (moveInfo && removalInfo && removeHappenedFirst && !moveIsLocal) {
+			// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+			const moveSeqOrLocalSeq = moveIsLocal ? moveInfo.localMovedSeq! : moveInfo.movedSeq;
```

Comment:
here and in the next if block you're already guaranteed to be working with a non-local change.
```suggestion
			const moveSeqOrLocalSeq = moveInfo.movedSeq;
```

(or can just inline the local; it doesn't do anything interesting at that point)